### PR TITLE
Report view

### PR DESCRIPTION
**Pull request by Omri-Levy**
- feat(backoffice-v2): added navlink and route for merchant monitoring (#2394)
- feat(backoffice-v2): added initial table with mocks - made generic no x and pagination components (#2397)
- Reports view backend integration (#2408)
- Reports view create report (#2424)
- feat(*): added route and link to a single report and added report id
- feat(backoffice-v2): added report view shell
- revert(backoffice-v2): reverted severity change from moderate to medium
- Report view shell (#2428)
- feat(backoffice-v2): mocked report summary
- fix(backoffice-v2): readded card header x padding
- feat(backoffice-v2): mocked all tabs of report view
- feat(*): integrated report view shell with backend
- feat(backoffice-v2): connected "view" links to tabs
- feat(backoffice-v2): mapped risk levels, summary, and recommendations
- feat(backoffice-v2): mapped risk indicators to data
- feat(*): checkpoint
- feat(backoffice-v2): no longer using mocks for company reputation analysis
- refactor(backoffice-v2): removed mcc related properties as they don't exist as of writing
- feat(backoffice-v2): added skeleton cards for website credibility
- feat(backoffice-v2): now passing real data to ecosystem table
- feat(backoffice-v2): connected all report ui to real data minus website credibility
- feat(backoffice-v2): added adapters, connected missing ui to real data
- feat(*): made merchant monitoring feature flagged
- feat(backoffice-v2): added conditions to start analyzing
- feat(backoffice-v2): populated missing data into report tabs

### CODE REVIEW

File: apps/backoffice-v2/src/domains/business-reports/adapters/report-adapter/report-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+  return label;
+};
+
+export const reportAdapter = {
+  DEFAULT: (data: Record<string, any>) => {
+    return {
+      websitesCompanyAnalysis:
+        data?.report?.data?.summary?.riskIndicatorsByDomain?.companyNameViolations?.map(
+          ({ name, riskLevel }) => ({
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/domains/business-reports/adapters/report-adapter/report-adapter.ts]: Re-use mapping function

File: apps/backoffice-v2/src/domains/business-reports/components/EcosystemAndTransactions/components/EcosystemAndTransactionsTable/columns.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+      return <TextWithNAFallback className={`ps-8`}>{index}</TextWithNAFallback>;
+    },
+    header: 'Match',
+  }),
+  columnHelper.accessor('matchedName', {
+    header: 'Matched Name',
+    cell: info => {
+      const matchedName = info.getValue();
+      const matchedNameWithProtocol = `http://${matchedName}`;
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/domains/business-reports/components/EcosystemAndTransactions/components/EcosystemAndTransactionsTable/columns.tsx]: Please check that the `matchedName` is not a URL before adding the protocol to it

File: apps/backoffice-v2/src/domains/business-reports/components/WebsiteLineOfBusiness/WebsiteLineOfBusiness.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      <Card>
+        <CardHeader className={'pt-4 font-bold'}>Line of Business Summary</CardHeader>
+        <CardContent className={'flex flex-col space-y-4'}>
+          <div>
+            <h4 className={'mb-4 font-semibold'}>LOB Description</h4>
+            <p>{summary}</p>
+          </div>
+        </CardContent>
+      </Card>
+      {/*<Card>*/}
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/domains/business-reports/components/WebsiteLineOfBusiness/WebsiteLineOfBusiness.tsx]: D3L3T3

File: apps/backoffice-v2/src/domains/customer/fetchers.ts [hunk 4]
```diff
@@ -18,6 +18,16 @@ export const fetchCustomer = async () => {
         customerStatus: z.string().optional(),
         country: z.union([z.string(), z.null()]).optional(),
         language: z.union([z.string(), z.null()]).optional(),
+        config: z
+          .object({
+            isMerchantMonitoringEnabled: z.boolean().default(false),
+            isExample: z.boolean().default(false),
+          })
+          .nullable()
+          .default({
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/domains/customer/fetchers.ts:27]: is the `default({...})` needed when we have defaults on the inner properties?

**Review by tomer-shvadron — APPROVED** [review]

File: apps/backoffice-v2/src/domains/customer/fetchers.ts (hunk 4 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/domains/customer/fetchers.ts:27]: Yes. You are supplying a default for when the root property is null or undefined.

**Pull request merged**